x-pack/plugins/observability/public/pages/alert_details/components/header_actions.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { noop } from 'lodash';
+import { CaseAttachmentsWithoutOwner } from '@kbn/cases-plugin/public/types';
+import { CommentType } from '@kbn/cases-plugin/common';
+import { EuiButton, EuiButtonEmpty, EuiFlexGroup, EuiPopover, EuiText } from '@elastic/eui';
+
+import { useKibana } from '../../../utils/kibana_react';
+import { useFetchRule } from '../../../hooks/use_fetch_rule';
+import { ObservabilityAppServices } from '../../../application/types';
+import { TopAlert } from '../../alerts';
+
+export interface HeaderActionsProps {
+  alert: TopAlert | null;
+  externalConnector?: {
+    name: string;
+    onViewInExternalApp: (alertId: string) => void;
+  };
+}
+
+export function HeaderActions({ alert, externalConnector }: HeaderActionsProps) {
+  const {
+    http,
+    cases: {
+      hooks: { getUseCasesAddToExistingCaseModal },
+    },
+    triggersActionsUi: { getEditAlertFlyout, getRuleSnoozeModal },
+  } = useKibana<ObservabilityAppServices>().services;
+
+  const { rule, reloadRule } = useFetchRule({
+    http,
+    ruleId: alert?.fields['kibana.alert.rule.uuid'] || '',
```

Comment:
We can use fields from `@kbn/rule-data-utils` in place of hardcoded field names.

Reply:
Ah nice. Will check!